### PR TITLE
Caching partitions metadata in memory.

### PR DESCRIPTION
**Pull request by OleksiiZubko**
This CR adds LRU cache support for the partitions metadata for the
versioned and volatile layers.
The user is able to set for each partition request param: "online-only" and "online if not found in cache".

By default for each request uses mode "online if not found in cache".
This mode decreases the count of requests to the services.

Disabling some tests. The tests will be enabled on the next CRs.

Resolves: OLPEDGE-1667

Signed-off-by: Oleksii Zubko <ext-oleksii.zubko@here.com>

### CODE REVIEW

File: @here/olp-sdk-dataservice-read/lib/client/FetchOptions.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
```
Comment (AndriyShevchuk) [inline @here/olp-sdk-dataservice-read/lib/client/FetchOptions.ts:2]: 2020

File: @here/olp-sdk-dataservice-read/lib/client/QueryClient.ts [hunk 2]
```diff
@@ -180,12 +197,33 @@ export class QueryClient {
...
             partition: idsList,
             additionalFields: request.getAdditionalFields(),
             version: version !== undefined ? `${version}` : undefined
-        });
+        }).catch(err => Promise.reject(err));
+
+        if (
+            request.getFetchOption() !== FetchOptions.OnlineOnly &&
+            medatada &&
+            medatada.partitions
```
Comment (AndriyShevchuk) [inline @here/olp-sdk-dataservice-read/lib/client/QueryClient.ts:211]: && metadata.partitions.length
In other case no sense to cache empty array

File: @here/olp-sdk-dataservice-read/lib/client/VersionedLayerClient.ts [hunk 3]
```diff
@@ -332,17 +334,40 @@ export class VersionedLayerClient {
...
             billingTag: request.getBillingTag()
-        });
+        }).catch(error => Promise.reject(error));
+
+        if (request.getFetchOption() !== FetchOptions.OnlineOnly) {
+            cache.put(
+                request,
+                this.hrn.toString(),
+                this.layerId,
+                metadata.partitions
```
Comment (AndriyShevchuk) [inline @here/olp-sdk-dataservice-read/lib/client/VersionedLayerClient.ts:367]: Is there any chance that metadata.partitions doesn't exist or is empty?

File: @here/olp-sdk-dataservice-read/lib/client/VolatileLayerClient.ts [hunk 4]
```diff
@@ -267,15 +271,39 @@ export class VolatileLayerClient {
...
             billingTag: request.getBillingTag()
-        });
+        }).catch(error => Promise.reject(error));
+
+        if (request.getFetchOption() !== FetchOptions.OnlineOnly) {
+            cache.put(
+                request,
+                this.hrn.toString(),
+                this.layerId,
+                metadata.partitions
```
Comment (AndriyShevchuk) [inline @here/olp-sdk-dataservice-read/lib/client/VolatileLayerClient.ts:303]: Is there any chance that metadata.partitions doesn't exist or is empty?

File: @here/olp-sdk-dataservice-read/lib/client/VersionedLayerClient.ts (hunk 3 above)
Comment (OleksiiZubko) [reply @here/olp-sdk-dataservice-read/lib/client/VersionedLayerClient.ts:367]: no

File: @here/olp-sdk-dataservice-read/lib/client/VolatileLayerClient.ts (hunk 4 above)
Comment (OleksiiZubko) [reply @here/olp-sdk-dataservice-read/lib/client/VolatileLayerClient.ts:303]: no

**Pull request merged**